### ISSUE TITLE
Getting No Application Key Specified even in artisan key:generate

### ISSUE THREAD

**Issue by eslym** [laravel/framework#33033]
- Laravel Version: v7.13.0 (6fa69bfbd57744a5bbec5538ce483919b3fd625f)
- PHP Version: 7.2.19
- Database Driver & Version:

### Description:
```cmd
C:\Users\xxx\Documents\projects\web\xxx>composer run post-create-project-cmd
> @php artisan key:generate --ansi

   RuntimeException 

  No application encryption key has been specified.

  at C:\Users\xxx\Documents\projects\web\xxx\vendor\laravel\framework\src\Illuminate\Encryption\EncryptionServiceProvider.php:80
    76|     protected function key(array $config)
    77|     {
    78|         return tap($config['key'], function ($key) {
    79|             if (empty($key)) {
  > 80|                 throw new RuntimeException(
    81|                     'No application encryption key has been specified.'
    82|                 );
    83|             }
    84|         });

  • Your app key is missing:
    https://laravel.com/docs/master/installation#configuration

  1   C:\Users\xxx\Documents\projects\web\xxx\vendor\laravel\framework\src\Illuminate\Support\helpers.php:433
      Illuminate\Encryption\EncryptionServiceProvider::Illuminate\Encryption\{closure}("")

  2   C:\Users\xxx\Documents\projects\web\xxx\vendor\laravel\framework\src\Illuminate\Encryption\EncryptionServiceProvider.php:84
      tap("", Object(Closure))
Script @php artisan key:generate --ansi handling the post-create-project-cmd event returned with error code 1

```
I guess this is not much to explain, just as what you guys can see.

p/s: I know I can generate a key from another project and use it but obviously this is a big issue.

### Steps To Reproduce:
1. clone a laravel project from git
2. copy `.env.example` to `.env`
3. run `php artisan key:generate`

**Comment by CaliforniaMountainSnake** [comment]
I saw same problems, `php artisan key:generate` doesn't work stable. Fifty-fifty, sometimes `APP_KEY` was set, sometimes not.
This behavior was on laravel 7, I didn't see this problem on laravel 5.8.
Now I setting this env-variable manually using external command:
```bash
php artisan -- env:set APP_KEY `php artisan key:generate --show`
```

**Comment by eslym** [comment]
> 
> 
> I saw same problems, `php artisan key:generate` doesn't work stable. Fifty-fifty, sometimes `APP_KEY` was set, sometimes not.
> This behavior was on laravel 7, I didn't see this problem on laravel 5.8.
> Now I setting this env-variable manually using external command:
> 
> ```shell
> php artisan -- env:set APP_KEY `php artisan key:generate --show`
> ```

then I guess we are having different problem, because I am having problem with **every** artisan command

**Comment by driesvints** [comment]
Hi there,

Thanks for reporting but it looks like this is a question which can be asked on a support channel. Please only use this issue tracker for reporting bugs with the library itself. If you have a question on how to use functionality provided by this repo you can try one of the following channels:

- [Laracasts Forums](https://laracasts.com/discuss)
- [Laravel.io Forums](https://laravel.io/forum)
- [StackOverflow](https://stackoverflow.com/questions/tagged/laravel)
- [Discord](https://discordapp.com/invite/KxwQuKb)
- [Larachat](https://larachat.co)
- [IRC](https://webchat.freenode.net/?nick=laravelnewbie&channels=%23laravel&prompt=1)

However, this issue will not be locked and everyone is still free to discuss solutions to your problem!

Thanks.

**Comment by buckhornmktg** [comment]
@driesvints this seems like a bug to me. My app key is specified and I get this behavior. Can't even run key:generate as described above. Can you either provide the solution or accept that this is a bug?

**Comment by devcircus** [comment]
Problem is that the team can't duplicate it. Personally, I just followed the instructions in the original post, and it worked exactly as it should. The OP did leave out the important step of "composer install", but I think that was just an oversight. Either way, when I try to duplicate the issue, everything works as it should. Usually these problems are due to either outdated packages, or permission issues on the machine being used. 

If you're using a different set of instructions than the OP, then this could be a different issue and needs its own issue thread.

git clone git@github.com:laravel/laravel.git
cd laravel
composer install
php artisan key:generate

**Comment by buckhornmktg** [comment]
I had the same issue as the OP and so did my dev. It's not ANY project, you would have to pull the same project to repeat the issue. I did run composer install, so that didn't seem to be the problem. I set permissions on .env to 0777, still no luck. 

The solution for me was to paste in an existing key from another project, then run php artisan key:generate. Until I pasted in another key, key:generate would not work.
